base/reducedim.jl

```diff
@@ -651,6 +651,8 @@ for (fname, _fname, op) in [(:sum,     :_sum,     :add_sum), (:prod,    :_prod,
         # User-facing methods with keyword arguments
         @inline ($fname)(a::AbstractArray; dims=:) = ($_fname)(a, dims)
         @inline ($fname)(f, a::AbstractArray; dims=:) = ($_fname)(f, a, dims)
+        @inline ($fname)(f, a::AbstractArray, b::AbstractArray; dims=:) =
+            ($_fname)(((a,b),)->f(a,b), zip(a,b), dims)
```

Comment:
If so, why don't make it more general? `sum(f, args...)` as `_sum(args->f(args...), zip(args...))`

Reply:
Good point!

Reply:
> If so, why don't make it more general? `sum(f, args...)` as `_sum(args->f(args...), zip(args...))`

This could even replace the single argument method, right?

Reply:
It could, but I will benchmark that first to ensure there's not a performance hit for the zip.